app/assets/javascripts/govuk_publishing_components/lib/header-navigation.js

```diff
@@ -7,12 +7,31 @@
 
   if (document.querySelectorAll && document.addEventListener) {
     var els = document.querySelectorAll('.js-header-toggle')
-    var i
-    var _i
-    for (i = 0, _i = els.length; i < _i; i++) {
-      els[i].addEventListener('click', function (e) {
+    for (var i = 0; i < els.length; i++) {
+      // Reassign current index to bypass rules in Chrome and Firefox around indexed property setting
+      var thisEl = els[i]
+
+      // If the element is an a tag, convert it to a button
+      // This is to target instances where we want to change behaviour between a js and a no-js view eg: the search button on the mobile menu for some government pages. On no-js it's just a link to /search
+      // This swaps a link for no-js to a button, making it interactable. Using a link for interactivity is poor accessibility as it's a misuse of the link tag and can be confusing to assistive tech users
+
+      if (thisEl.tagName === 'A') {
+        var button = document.createElement('button')
+        var href = thisEl.getAttribute('href')
+        var buttonText = thisEl.getAttribute('data-button-text')
+
+        button.className = thisEl.className
+
+        button.setAttribute('data-search-toggle-for', href ? href.substr(1) : thisEl.getAttribute('data-search-toggle-for'))
+        button.innerText = buttonText || thisEl.innerText
+
+        thisEl.parentNode.replaceChild(button, thisEl)
+        thisEl = button
+      }
```

Comment:
Any other attributes that might have to be copied over in the future (like tracking attributes)?

Reply:
That's a good shout. Ideally we'd take everything and maybe do some clean-up. Will investigate.

Reply:
Reckon I've found a solution. Let me know what you think. Being bold and resolving this for now.